classifier/classifier.py

```diff
@@ -1,138 +1,247 @@
+            return False
+        if self.args.types:
+            # Show file format information then quit
+            for key, value in self.formats.items():
+                print(key, '\n', value)
+            return False
+        if self.args.edittypes:
+            subprocess.Popen(['xdg-open', CONFIG])
+            return False
+        if bool(self.args.specific_folder) ^ bool(self.args.specific_types):
+            print(
+                'Specific Folder and Specific Types need to be specified together')
+            sys.exit()
+
+        if self.args.specific_folder and self.args.specific_types:
+            specific_folder = self._format_arg(self.args.specific_folder)
+            self.formats = {specific_folder: self.args.specific_types}
+
+        if self.args.output is None:
+            output = os.getcwd()
+        else:
+            output = self._format_arg(self.args.output)
+
+        if self.args.directory is None:
+            directory = os.getcwd()
+        else:
+            directory = self._format_arg(self.args.directory)
+            if self.args.output is None:
+                ''' if -d arg given without the -o arg, keeping the files of -d
+                in the -d path only after classifying '''
+                output = directory
+
+        # Check for a config file in the source file directory
+        if self.args.directory:
+            if os.path.isfile(os.path.join(self.args.directory, DIRCONFFILE)):
+                self.dirconf = os.path.join(self.args.directory, DIRCONFFILE)
+        elif os.path.isfile(os.path.join(os.getcwd(), DIRCONFFILE)):
+            self.dirconf = os.path.join(os.getcwd(), DIRCONFFILE)
+
+        if self.args.date:
+            if self.args.dateformat:
+                self.classify_by_date(self.args.dateformat, output, directory)
+            else:
+                self.classify_by_date(self.dateformat, output, directory)
+        elif os.path.isfile(self.dirconf):
+            print('Found config in current directory')
+            if self.args.output:
+                print('Your output directory is being ignored!!!')
+            for items in open(self.dirconf, "r"):
+                # reset formats for individual folders
+                self.formats = {}
+                (key, dst, val) = items.split(':')
+                self.formats[key] = val.replace('\n', '').split(',')
+                print("\nScanning:  " + directory +
+                      "\nFor:       " + key +
+                      '\nRecursive: ' + str(self.args.recursive) +
+                      '\nFormats:   ' + val)
+                self.classify(self.formats, dst, directory)
+        else:
+            print("\nScanning Folder: " + directory +
+                  "\nFor: " + str(self.formats.items()) +
+                  '\nRecursive: ' + str(self.args.recursive))
+            self.classify(self.formats, output, directory)
+
+        print("Done!\n")
+        return True
+
+
+Classifier()
```

Comment:
since line 14 of `setup.py` calls for a `main` function, put this initialization inside a function named main:
``` python
def main():
    Classifier()
```
(this was done for Windows compatibility)
And you can rename the member function main of the Classifier class to smth like `run`